src/hevm/src/EVM.hs

```diff
@@ -906,6 +910,33 @@ exec1 = do
         -- op: JUMPDEST
         0x5b -> burn g_jumpdest next
 
+        -- op: BEGINSUB
+        0x5c -> vmError InvalidSubroutineEntry
+
+        -- op: RETURNSUB
+        0x5d -> case the state return_stack of
+            (x:xs) -> burn g_low $ do
+                state . pc .= x
+                state . return_stack .= xs
+            _ -> underrun
+
+        -- op: JUMPSUB
+        0x5e ->
+          case stk of
+            (x:xs) ->
+              burn g_high $ do
+                theCode <- use (state . code)
+                theReturnStack <- use (state . return_stack)
+                thePc <- use (state . pc)
+                if x < num (BS.length theCode) && BS.index theCode (num x) == 0x5c
```

Comment:
Don't you need to make sure that this `0x5c` is not inside a push? Cf. the `checkJump` function and the `codeOps` checking

Reply:
aha! It was not at all clear to me what the `codeOps` check was there for. Thx :hugs:

Reply:
Is there a reason why `checkJump` checks for `0x5b` in `theCode` and also for `OpJumpdest` in `theCodeOps`? 

Afaict the first check (in `theCode`) is not required if the second (in `theCodeOps`) is present?

Reply:
yeah, you're right. Might be an optimization, but I'm not sure

Reply:
`JUMPSUB` should now reject jumps to pushdata. I also factored the `theCodeOps` lookup into a helper and removed the duplicated check in `checkJump`.